### PR TITLE
Switching from jshint/jscs to ESLint

### PR DESCRIPTION
**Pull request by jbeezley**
ESLint make validating commonjs semantics much more pleasant so I'm making this transition now help help with the on going work in #515.  The rules included are relatively minimal for the moment so that I don't have to go through and make a bunch of manual code changes.  The majority of the code changes in this PR were done automatically using `eslint --fix` which can correct simple errors such as indentation and `'` vs. `"` issues.  I decided to get all of those changes out of the way now so we can just set the rules and move on.

### CODE REVIEW

**Comment by aashish24** [comment]
Nice !

File: src/gl/vglRenderer.js [hunk 1]
```diff
@@ -249,7 +249,7 @@ geo.registerRenderer('vgl', geo.gl.vglRenderer);
   geo.gl.vglRenderer.supported = function () {
     if (checkedWebGL === undefined) {
       /* This is extracted from what Modernizr uses. */
-      var canvas, ctx, exts;
+      var canvas, ctx, exts; // eslint-disable-line no-unused-vars
```
Comment (aashish24) [inline src/gl/vglRenderer.js]: I am wondering about "eslint-disable-line no-unused-vars" here.

**Comment by aashish24** [comment]
Looks good to me :+1:

File: src/gl/vglRenderer.js (hunk 1 above)
Comment (jbeezley) [inline src/gl/vglRenderer.js]: It's a special case.  The variables are used in the try/catch block below.  It's possible we could get rid of them, but I didn't think it worth my time figuring out.

Comment (aashish24) [inline src/gl/vglRenderer.js]: Thanks! I was wondering about that.  LGTM :+1:

**Comment by jbeezley** [comment]
@manthey I manually fixed some of the weird issues that you brought up.

**Comment by manthey** [comment]
LGTMWTIH

**Pull request merged**